### ISSUE TITLE
Geth will not terminate on Windows x64

### ISSUE THREAD

**Issue by mrbenfraser** [ethereum/go-ethereum#2203]
Seemed to do nothing for many hours so I ctrl-C'd.

Not terminated after a further half hour. Using 0 CPU (so not a duplicate of #1892) and memory occupancy per task manager is not deceasing (162MB) so I decided to Ctrl-C 9 more times. Output below.

```
I0211 03:16:14.901927    9776 blockchain.go:1243] imported 1 block(s) (0 queued 0 ignored) including 1 txs in 7.0004ms. #985779 [dd43f607 /
dd43f607]
I0211 20:02:31.733284    9776 cmd.go:123] Got interrupt, shutting down...
I0211 20:04:45.018907    9776 cmd.go:129] Already shutting down, please be patient.
I0211 20:04:45.018907    9776 cmd.go:130] Interrupt 9 more times to induce panic.
I0211 20:28:50.990612    9776 cmd.go:129] Already shutting down, please be patient.
I0211 20:28:50.990612    9776 cmd.go:130] Interrupt 8 more times to induce panic.
I0211 20:34:35.660326    9776 cmd.go:129] Already shutting down, please be patient.
I0211 20:34:35.660326    9776 cmd.go:130] Interrupt 7 more times to induce panic.
I0211 20:34:38.388482    9776 cmd.go:129] Already shutting down, please be patient.
I0211 20:34:38.388482    9776 cmd.go:130] Interrupt 6 more times to induce panic.
I0211 20:34:39.108524    9776 cmd.go:129] Already shutting down, please be patient.
I0211 20:34:39.108524    9776 cmd.go:130] Interrupt 5 more times to induce panic.
I0211 20:34:39.668556    9776 cmd.go:129] Already shutting down, please be patient.
I0211 20:34:39.668556    9776 cmd.go:130] Interrupt 4 more times to induce panic.
I0211 20:34:40.188585    9776 cmd.go:129] Already shutting down, please be patient.
I0211 20:34:40.188585    9776 cmd.go:130] Interrupt 3 more times to induce panic.
I0211 20:34:40.764618    9776 cmd.go:129] Already shutting down, please be patient.
I0211 20:34:40.764618    9776 cmd.go:130] Interrupt 2 more times to induce panic.
I0211 20:34:41.452658    9776 cmd.go:129] Already shutting down, please be patient.
I0211 20:34:41.452658    9776 cmd.go:130] Interrupt 1 more times to induce panic.
I0211 20:34:42.180699    9776 cmd.go:133] Force quitting: this might not end so well.
panic: boom

goroutine 84 [running]:
github.com/ethereum/go-ethereum/cmd/utils.StartEthereum.func1(0xc082272140)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/cmd/utils/cmd.go:134 +0x6bf
created by github.com/ethereum/go-ethereum/cmd/utils.StartEthereum
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/cmd/utils/cmd.go:135 +0x2d7

goroutine 1 [chan receive, 3154 minutes]:
main.run(0xc082155680)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:418 +0x204
github.com/codegangsta/cli.(*App).Run(0xc082155560, 0xc082006300, 0x4, 0x4, 0x0, 0x0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:180 +0x102f
main.main()
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:373 +0x9a

goroutine 17 [syscall, 3154 minutes, locked to thread]:
runtime.goexit()
        c:/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
        c:/go/src/os/signal/signal_unix.go:22 +0x1f
created by os/signal.init.1
        c:/go/src/os/signal/signal_unix.go:28 +0x3e

goroutine 6 [chan receive, 2 minutes]:
github.com/ethereum/go-ethereum/logger/glog.(*loggingT).flushDaemon(0x1512700)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/logger/glog/glog.go:897 +0x6e
created by github.com/ethereum/go-ethereum/logger/glog.init.1
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/logger/glog/glog.go:428 +0x76

goroutine 7 [select, 32 minutes]:
github.com/ethereum/go-ethereum/logger.dispatchLoop()
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/logger/sys.go:81 +0x62b
created by github.com/ethereum/go-ethereum/logger.init.1
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/logger/sys.go:60 +0x32

goroutine 85 [syscall, 2777 minutes, locked to thread]:
syscall.Syscall(0x770f1530, 0x2, 0x2f0, 0xffffffff, 0x0, 0xc08373fdb0, 0xc0820c4808, 0x3c0050)
        c:/go/src/runtime/syscall_windows.go:128 +0x5c
syscall.WaitForSingleObject(0x2f0, 0xc0ffffffff, 0xc08373fdd0, 0x0, 0x0)
        c:/go/src/syscall/zsyscall_windows.go:690 +0x6f
github.com/ethereum/go-ethereum/rpc/comms.waitForCompletion(0x1c8, 0xc0993c9fa0, 0xc083798140, 0x0, 0x0)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/rpc/comms/ipc_windows.go:314 +0x45
github.com/ethereum/go-ethereum/rpc/comms.(*PipeListener).AcceptPipe(0xc083798140, 0x0, 0x0, 0x0)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/rpc/comms/ipc_windows.go:446 +0x397
github.com/ethereum/go-ethereum/rpc/comms.(*PipeListener).Accept(0xc083798140, 0x0, 0x0, 0x0, 0x0)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/rpc/comms/ipc_windows.go:394 +0x44
github.com/ethereum/go-ethereum/rpc/comms.ipcLoop(0xf263b0, 0x11, 0x0, 0xc0825840c0, 0x481a8b0, 0xc083798140)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/rpc/comms/ipc.go:116 +0x1b1
created by github.com/ethereum/go-ethereum/rpc/comms.StartIpc
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/rpc/comms/ipc.go:107 +0xd8

goroutine 83 [select]:
github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc08223ce00)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:205 +0x
278
created by github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:236 +0x
272

goroutine 68 [select, 701 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc0822306e0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:120 +0x483

created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:126 +0x803

goroutine 807391 [chan receive, 32 minutes]:
github.com/ethereum/go-ethereum/logger.sysLoop(0x3c9850, 0xc082203180, 0xc08f4e4060, 0xc0820c0000)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/logger/sys.go:117 +0x58
created by github.com/ethereum/go-ethereum/logger.dispatchLoop.func1
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/logger/sys.go:77 +0xe2

goroutine 50 [select]:
github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc08223c0e0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:205 +0x
278
created by github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:236 +0x
272

goroutine 12 [select, 1053 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc0822402c0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:120 +0x483

created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:126 +0x803

goroutine 13 [select, 2 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc0822402c0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_state.go:82 +0x152
created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:127 +0x825

goroutine 14 [select, 1053 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc0822402c0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:812 +0x7cf

created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:130 +0x871

goroutine 15 [select, 1061 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc0822402c0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:759 +0x25a

created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:131 +0x893

goroutine 16 [select, 3154 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).jWriter(0xc0822402c0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_write.go:37 +0x1a9
created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:132 +0x8b5

goroutine 66 [select]:
github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).meter(0xc08206e000, 0xb2d05e00)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/ethdb/database.go:255 +0xa9d
created by github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).Meter
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/ethdb/database.go:187 +0x598

goroutine 67 [select]:
github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc08222c0e0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:205 +0x
278
created by github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:236 +0x
272

goroutine 39 [select, 3154 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc082240580)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:120 +0x483

created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:126 +0x803

goroutine 40 [select, 2 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc082240580)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_state.go:82 +0x152
created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:127 +0x825

goroutine 41 [select, 3154 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc082240580)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:812 +0x7cf

created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:130 +0x871

goroutine 42 [select, 3154 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc082240580)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:759 +0x25a

created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:131 +0x893

goroutine 43 [select, 3154 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).jWriter(0xc082240580)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_write.go:37 +0x1a9
created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:132 +0x8b5

goroutine 44 [select]:
github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).meter(0xc08206e0c0, 0xb2d05e00)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/ethdb/database.go:255 +0xa9d
created by github.com/ethereum/go-ethereum/ethdb.(*LDBDatabase).Meter
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/ethdb/database.go:187 +0x598

goroutine 45 [select]:
github.com/ethereum/go-ethereum/core.(*BlockChain).update(0xc082272280)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/core/blockchain.go:1362 +0x108
created by github.com/ethereum/go-ethereum/core.NewBlockChain
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/core/blockchain.go:174 +0x148b

goroutine 46 [chan receive, 1038 minutes]:
github.com/ethereum/go-ethereum/core.(*TxPool).eventLoop(0xc0820ea150)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/core/transaction_pool.go:95 +0xa9
created by github.com/ethereum/go-ethereum/core.NewTxPool
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/core/transaction_pool.go:86 +0x3b3

goroutine 47 [select, 1038 minutes]:
github.com/ethereum/go-ethereum/miner.(*worker).update(0xc082408000)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/miner/worker.go:228 +0x6f8
created by github.com/ethereum/go-ethereum/miner.newWorker
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/miner/worker.go:142 +0x3c5

goroutine 48 [chan receive, 3154 minutes]:
github.com/ethereum/go-ethereum/miner.(*worker).wait(0xc082408000)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/miner/worker.go:271 +0x9a
created by github.com/ethereum/go-ethereum/miner.newWorker
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/miner/worker.go:143 +0x3e7

goroutine 82 [select, 6 minutes]:
github.com/ethereum/go-ethereum/p2p/nat.Map(0x3c9810, 0xc0822045c0, 0xc082526120, 0xe93e18, 0x3, 0x765f, 0x765f, 0xf2ae90, 0x12)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/nat/nat.go:114 +0xaa1
created by github.com/ethereum/go-ethereum/p2p/discover.newUDP
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:219 +0x2b9

goroutine 69 [select, 2 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc0822306e0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_state.go:82 +0x152
created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:127 +0x825

goroutine 70 [select, 701 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc0822306e0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:812 +0x7cf

created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:130 +0x871

goroutine 71 [select, 701 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc0822306e0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_compaction.go:759 +0x25a

created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:131 +0x893

goroutine 72 [select, 3154 minutes]:
github.com/syndtr/goleveldb/leveldb.(*DB).jWriter(0xc0822306e0)
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db_write.go:37 +0x1a9
created by github.com/syndtr/goleveldb/leveldb.openDB
        D:/builds/geth_win32/build-go-win-master/build/Godeps/_workspace/src/github.com/syndtr/goleveldb/leveldb/db.go:132 +0x8b5

goroutine 73 [chan receive, 32 minutes]:
github.com/ethereum/go-ethereum/p2p/discover.(*Table).refreshLoop(0xc0820a2900)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:303 +0x280
created by github.com/ethereum/go-ethereum/p2p/discover.newTable
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:116 +0x611

goroutine 74 [select, 32 minutes]:
github.com/ethereum/go-ethereum/p2p/discover.(*udp).loop(0xc08251e850)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:341 +0xb29
created by github.com/ethereum/go-ethereum/p2p/discover.newUDP
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:229 +0x516

goroutine 78 [chan receive, 32 minutes]:
github.com/ethereum/go-ethereum/p2p/discover.(*Table).Close(0xc0820a2900)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:179 +0x103
github.com/ethereum/go-ethereum/p2p.(*Server).run(0xc082079c80, 0x3cb110, 0xc08226a230)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/server.go:504 +0x621
created by github.com/ethereum/go-ethereum/p2p.(*Server).Start
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/server.go:362 +0xce9

goroutine 79 [chan receive, 1038 minutes]:
github.com/ethereum/go-ethereum/eth.(*ProtocolManager).txBroadcastLoop(0xc08206e300)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/handler.go:742 +0x86
created by github.com/ethereum/go-ethereum/eth.(*ProtocolManager).Start
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/handler.go:179 +0x121

goroutine 80 [chan receive, 3154 minutes]:
github.com/ethereum/go-ethereum/eth.(*ProtocolManager).minedBroadcastLoop(0xc08206e300)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/handler.go:731 +0x8d
created by github.com/ethereum/go-ethereum/eth.(*ProtocolManager).Start
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/handler.go:182 +0x21e

goroutine 81 [select]:
github.com/ethereum/go-ethereum/eth.(*ProtocolManager).syncer(0xc08206e300)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/sync.go:139 +0x267
created by github.com/ethereum/go-ethereum/eth.(*ProtocolManager).Start
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/handler.go:185 +0x243

goroutine 98 [select, 1546 minutes]:
github.com/ethereum/go-ethereum/eth.(*ProtocolManager).txsyncLoop(0xc08206e300)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/sync.go:105 +0x60f
created by github.com/ethereum/go-ethereum/eth.(*ProtocolManager).Start
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/handler.go:186 +0x268

goroutine 99 [select, 1038 minutes]:
github.com/ethereum/go-ethereum/eth/fetcher.(*Fetcher).loop(0xc082e60dd0)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/fetcher/fetcher.go:350 +0x6839
created by github.com/ethereum/go-ethereum/eth/fetcher.(*Fetcher).Start
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/fetcher/fetcher.go:182 +0x3c

goroutine 21 [select, 34 minutes]:
github.com/ethereum/go-ethereum/p2p/discover.(*nodeDB).expirer(0xc082526cc0)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/database.go:233 +0x231
created by github.com/ethereum/go-ethereum/p2p/discover.(*nodeDB).ensureExpirer.func1
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/database.go:225 +0x3b

goroutine 198373 [chan send, 2801 minutes]:
github.com/ethereum/go-ethereum/rpc/comms.(*PipeConn).completeRequest.func1(0xc08c83c0c0, 0xc0921bd740, 0xc089825020)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/rpc/comms/ipc_windows.go:532 +0x8f
created by github.com/ethereum/go-ethereum/rpc/comms.(*PipeConn).completeRequest
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/rpc/comms/ipc_windows.go:533 +0x17d

goroutine 807415 [select, 32 minutes]:
github.com/ethereum/go-ethereum/p2p/discover.(*Table).requestRefresh(0xc0820a2900)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:270 +0xe4
github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup(0xc0820a2900, 0xf87f466c62aa80e4, 0xaf69fd24eaabe19c, 0xd9ece5d7df7075c7, 0x7f3
e0fe12737de86, 0x38d9743319eca3dd, 0x7fbd227ac14ad47c, 0x39cb7c7a10f53128, 0x82befbb0425384b6, 0x0, ...)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:224 +0x372
github.com/ethereum/go-ethereum/p2p/discover.(*Table).doRefresh(0xc0820a2900, 0xc08e0faf60)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:348 +0x665
created by github.com/ethereum/go-ethereum/p2p/discover.(*Table).refreshLoop
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:297 +0x1f2

goroutine 807414 [select, 32 minutes]:
github.com/ethereum/go-ethereum/p2p/discover.(*Table).requestRefresh(0xc0820a2900)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:270 +0xe4
github.com/ethereum/go-ethereum/p2p/discover.(*Table).Lookup(0xc0820a2900, 0xd46d55e38157941b, 0x15c1a3b26992578c, 0x9b83191c187851ae, 0x32f
328873c2000f8, 0x929ee8d0aa7eb523, 0x9c34e2d9b6ec9200, 0xe3a2e51a4acfd295, 0xe0b8f30b581df494, 0x0, ...)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/discover/table.go:224 +0x372
github.com/ethereum/go-ethereum/p2p.(*discoverTask).Do(0xc09220e2c0, 0xc082079c80)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/dial.go:238 +0x32b
github.com/ethereum/go-ethereum/p2p.(*Server).run.func2.1(0x4857600, 0xc09220e2c0, 0xc082079c80, 0xc082514060)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x38
created by github.com/ethereum/go-ethereum/p2p.(*Server).run.func2
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/server.go:432 +0x5ba

goroutine 242877 [chan receive, 1038 minutes]:
github.com/ethereum/go-ethereum/eth.(*GasPriceOracle).listenLoop(0xc08a1045a0)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/gasprice.go:107 +0x253
created by github.com/ethereum/go-ethereum/eth.(*GasPriceOracle).init.func1
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/gasprice.go:79 +0x5a

goroutine 807390 [semacquire, 32 minutes]:
sync.runtime_Semacquire(0xc082079df4)
        c:/go/src/runtime/sema.go:43 +0x2d
sync.(*WaitGroup).Wait(0xc082079de8)
        c:/go/src/sync/waitgroup.go:126 +0xbb
github.com/ethereum/go-ethereum/p2p.(*Server).Stop(0xc082079c80)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/p2p/server.go:299 +0x108
github.com/ethereum/go-ethereum/eth.(*Ethereum).Stop(0xc082272140)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/eth/backend.go:550 +0x33
created by github.com/ethereum/go-ethereum/cmd/utils.StartEthereum.func1
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/cmd/utils/cmd.go:124 +0x269

goroutine 242817 [chan send, 2776 minutes]:
github.com/ethereum/go-ethereum/rpc/comms.(*PipeConn).completeRequest.func1(0xc085556000, 0xc09fac0660, 0xc099e141e0)
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/rpc/comms/ipc_windows.go:532 +0x8f
created by github.com/ethereum/go-ethereum/rpc/comms.(*PipeConn).completeRequest
        D:/builds/geth_win32/build-go-win-master/go/src/github.com/ethereum/go-ethereum/rpc/comms/ipc_windows.go:533 +0x17d

D:\Ethereum-Wallet-win32-x64-0-3-9\resources\node\geth>
```

**Comment by fjl** [comment]
This issue was fixed "by accident" in #1925, which will land in geth 1.4.

**Comment by amacneil** [comment]
Same problem on OSX. Is there an easy workaround?
